scala/src/main/org/apache/spark/api/csharp/CSharpBackendHandler.scala

```diff
@@ -11,21 +11,31 @@ import java.net.Socket
 
 import io.netty.channel.ChannelHandler.Sharable
 import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
+
 // TODO - work with SparkR devs to make this configurable and reuse RBackendHandler
 import org.apache.spark.api.csharp.SerDe._
 
 import scala.collection.mutable.HashMap
 
 /**
- * Handler for CSharpBackend.
- * This implementation is identical to RBackendHandler and that can be reused
- * in SparkCLR if SerDe is made pluggable
- */
+  * Handler for CSharpBackend.
+  * This implementation is identical to RBackendHandler and that can be reused
+  * in SparkCLR if SerDe is made pluggable
+  */
 // Since SparkCLR is a package to Spark and not a part of spark-core, it mirrors the implementation
 // of selected parts from RBackend with SparkCLR customizations
 class CSharpBackendHandler(server: CSharpBackend) extends SimpleChannelInboundHandler[Array[Byte]] {
 
   override def channelRead0(ctx: ChannelHandlerContext, msg: Array[Byte]): Unit = {
+    val reply = readChannelMessage(msg)
+    ctx.write(reply)
+  }
+
+  override def channelReadComplete(ctx: ChannelHandlerContext): Unit = {
+    ctx.flush()
+  }
+
+  def readChannelMessage(msg: Array[Byte]): Array[Byte] = {
```

Comment:
What is the reason for this new method? The name of the method is misleading because it is not just reading the message but also processing it. Consider reverting this change or rename the method.